### DISCUSSION TITLE
MAC spoofing

### DISCUSSION THREAD

**Discussion by karolborychowski** [strongswan/strongswan#857] [Q&A] [answered]
Hello everyone,
I hope I am not duplicating any already existing topic but I have really weird situation. It seems when I enable my strongswan setup, it starts spoofing MAC addresses by answering ARP requests. 
Here is a snippet of ipsec.conf (the legacy configuration file).
       type=tunnel
        auto=start
        keyexchange=ikev1
        left=*********
        leftsubnet=0.0.0.0/0
        leftfirewall=yes
        right=***********
        rightid=***********
        rightsubnet=0.0.0.0/0
        ike=aes256-sha1-modp1024!
        esp=aes256-sha1-modp1024!
        aggressive=no
        keyingtries=%forever
        ikelifetime=86400s
        lifetime=3600s
        dpddelay=30s
        dpdtimeout=120s
        dpdaction=restart
        leftid=***********
        leftcert=*************
        mark=104
I am using a routed VPN with VTI interfaces.
strongswan --version
Linux strongSwan U5.7.2/K3.10.0-957.27.2.el7.x86_64
University of Applied Sciences Rapperswil, Switzerland
See 'strongswan --copyright' for copyright information.

**Comment by karolborychowski** [comment, marked as answer]
Update: my current focus is on strongswan.d/charon/farp.conf - it seems as something dedicated to fake ARP replies.
I keep observing.

**Reply by tobiasbrunner** [reply]
It is, see [FARPPlugin](https://wiki.strongswan.org/projects/strongswan/wiki/FARPPlugin).

**Reply by karolborychowski** [reply]
Thank you. That was it.
Honestly, a very dangerous feature enabled by default when running in a data centre. 
I installed from a package.

Regards,

**Reply by tobiasbrunner** [reply]
> I installed from a package.

You might want to report this to the package maintainers (we don't maintain any binary packages ourselves).
